Compilation error with `-mcpu=a64fx` on Ookami

## Expected Behavior

Compiled successfully.

## Actual Behavior

I got this error.
```
/XXX/workspace/hpx/libs/core/coroutines/include/hpx/coroutines/detail/context_linux_x86.hpp: In member function ‘void hpx::threads::coroutines::detail::lx::x86_linux_context_impl_base::prefetch() const’:
/XXX/workspace/hpx/libs/core/coroutines/include/hpx/coroutines/detail/context_linux_x86.hpp:206:41: error: static assertion failed
  206 |             static_assert(sizeof(void*) == 4);
      |                           ~~~~~~~~~~~~~~^~~~
ninja: build stopped: subcommand failed.
[jiakyan@login1 install_scripts]$ ^C
```

## Steps to Reproduce the Problem

I was compiling HPX on Ookami.

The dependencies loaded in the environment were:
```
gcc/11.2.0 cmake/3.24.2 boost/1.71.0 hwloc/2.8.0 openmpi/gcc11.2/4.1.2 jemalloc/5.3.0 Vc/1.4.3 papi/6.0.0
```

I used the following commands to configure the build
```
export CC=gcc
export CXX=g++
export CFLAGS="-mcpu=a64fx"
export CXXFLAGS="-mcpu=a64fx"
cmake -H${SRC_PATH} \
    -B${BUILD_PATH} \
    -DCMAKE_INSTALL_PREFIX=${INSTALL_PATH} \
    -DCMAKE_C_COMPILER=${CC} \
    -DCMAKE_CXX_COMPILER=${CXX} \
    -DCMAKE_C_FLAGS="${CFLAGS}" \
    -DCMAKE_CXX_FLAGS="${CXXFLAGS}" \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_VERBOSE_MAKEFILE=ON \
    -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
    -GNinja \
    -DHPX_WITH_COMPILER_WARNINGS=ON \
    -DHPX_WITH_COMPILER_WARNINGS_AS_ERRORS=ON \
    -DHPX_WITH_FETCH_ASIO=ON\
    -DHPX_WITH_DISABLED_SIGNAL_EXCEPTION_HANDLERS=ON \
    -DHPX_WITH_MALLOC=JEMALLOC \
    -DHPX_WITH_MAX_CPU_COUNT=256 \
    -DHPX_WITH_EXAMPLES=OFF \
    -DHPX_WITH_TESTS=OFF \
    -DHPX_WITH_NETWORKING=ON \
    -DHPX_WITH_PARCELPORT_MPI=ON \
    -DHPX_WITH_PARCELPORT_LCI=ON \
    -DHPX_WITH_ZERO_COPY_SERIALIZATION_THRESHOLD=8192 \
    -DHPX_WITH_CUDA=${GIS_WITH_CUDA} \
    -DAPEX_WITH_ACTIVEHARMONY=FALSE \
    -DAPEX_WITH_MPI=ON \
    -DHPX_WITH_CXX_STANDARD=20 \
    -DHPX_WITH_DATAPAR_BACKEND=STD_EXPERIMENTAL_SIMD
```

## Specifications

  - HPX Version: (Almost) The current master branch (066d16cec3427f4a2655ea4ab5970d833f351588).
  - Platform (compiler, OS): GCC 11.2, Rocky Linux 8.4 (Green Obsidian)

I looked at the code and found these lines.

`libs/core/coroutines/include/hpx/coroutines/detail/context_impl.hpp`
```
#if defined(HPX_HAVE_GENERIC_CONTEXT_COROUTINES)
...
#elif (defined(__linux) || defined(linux) || defined(__linux__)) &&            \
    !defined(__bgq__) && !defined(__powerpc__) && !defined(__s390x__)

#include <hpx/coroutines/detail/context_linux_x86.hpp>
```

`libs/core/coroutines/include/hpx/coroutines/detail/context_linux_x86.hpp`
```
#if defined(__x86_64__)
            static_assert(sizeof(void*) == 8);
#else
            static_assert(sizeof(void*) == 4);
#endif
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1582,8 +1582,8 @@ endif()
 if("${HPX_PLATFORM_UC}" STREQUAL "BLUEGENEQ")
   set(__use_generic_coroutine_context ON)
 endif()
-# If compiling for riscv64, automatically bake in Boost.Context
-if(${__target_arch} STREQUAL "riscv64")
+# If compiling for riscv64 or arm64, automatically bake in Boost.Context
+if(${__target_arch} STREQUAL "riscv64" OR ${__target_arch} STREQUAL "arm64")
   set(__use_generic_coroutine_context ON)
 endif()
 

